racket/src/ChezScheme/rktboot/make-boot.rkt

```diff
@@ -255,7 +262,7 @@
               [(constant-case architecture [else e ...])
                (loop #`(begin e ...))]
               [(constant-case architecture [(arch ...) e ...] . _)
-               (memq (string->symbol target-machine) (syntax->datum #'(arch ...)))
+               (memq target-arch (syntax->datum #'(arch ...)))
```

Comment:
I'll push a more general repair here, which is to use the `architecture` constant as defined earlier by the machine's ".def" file.